File: src/calculation.php
<?php
echo "Welcome to the Inefficient PHP Application!<br>";

// Unnecessary delay
sleep(2);

echo "Enter the first number: ";
$num1 = 10; // Simulating user input
echo $num1 . "<br>";

// Unnecessary loop
for ($i = 0; $i < 1000000; $i++) {
    $num1 += 0.000001;
}

echo "Enter the second number: ";
$num2 = 20; // Simulating user input
echo $num2 . "<br>";

// Unnecessary loop
for ($i = 0; $i < 1000000; $i++) {
    $num2 += 0.000001;
}

echo "Choose an operation:<br>";
echo "1. Addition<br>";
echo "2. Subtraction<br>";
echo "3. Multiplication<br>";
echo "4. Division<br>";

$choice = 1; // Simulating user input
echo "You chose: " . $choice . "<br>";

$result = 0;

// Inefficient data structure
$operations = array(
    1 => 'add',
    2 => 'subtract',
    3 => 'multiply',
    4 => 'divide'
);

if (array_key_exists($choice, $operations)) {
    $result = $operations$choice;
} else {
    echo "Invalid choice.<br>";
}

// Unnecessary delay
sleep(2);

echo "The result is: " . $result . "<br>";

function add($a, $b) {
    // Redundant computation
    for ($i = 0; $i < 1000000; $i++) {}
    return $a + $b;
}

function subtract($a, $b) {
    // Redundant computation
    for ($i = 0; $i < 1000000; $i++) {}
    return $a - $b;
}

function multiply($a, $b) {
    // Redundant computation
    for ($i = 0; $i < 1000000; $i++) {}
    return $a * $b;
}

function divide($a, $b) {
    // Redundant computation
    for ($i = 0; $i < 1000000; $i++) {}
    if ($b != 0) {
        return $a / $b;
    } else {
        echo "Cannot divide by zero.<br>";
        return 0;
    }
}
?>

File: src/calculation.cs
using System;
using System.Collections.Generic;
using System.Threading;

namespace SampleDotNetApp
{
    class InefficientProgram
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Welcome to the Inefficient .NET Application!");

            // Unnecessary delay
            Thread.Sleep(2000);

            Console.Write("Enter the first number: ");
            double num1 = Convert.ToDouble(Console.ReadLine());

            // Unnecessary loop
            for (int i = 0; i < 1000000; i++)
            {
                num1 += 0.000001;
            }

            Console.Write("Enter the second number: ");
            double num2 = Convert.ToDouble(Console.ReadLine());

            // Unnecessary loop
            for (int i = 0; i < 1000000; i++)
            {
                num2 += 0.000001;
            }

            Console.WriteLine("Choose an operation:");
            Console.WriteLine("1. Addition");
            Console.WriteLine("2. Subtraction");
            Console.WriteLine("3. Multiplication");
            Console.WriteLine("4. Division");

            int choice = Convert.ToInt32(Console.ReadLine());

            double result = 0;

            // Inefficient data structure
            Dictionary<int, Func<double, double, double>> operations = new Dictionary<int, Func<double, double, double>>
            {
                { 1, Add },
                { 2, Subtract },
                { 3, Multiply },
                { 4, Divide }
            };

            if (operations.ContainsKey(choice))
            {
                result = operationschoice;
            }
            else
            {
                Console.WriteLine("Invalid choice.");
            }

            // Unnecessary delay
            Thread.Sleep(2000);

            Console.WriteLine($"The result is: {result}");
        }

        static double Add(double a, double b)
        {
            // Redundant computation
            for (int i = 0; i < 1000000; i++) { }
            return a + b;
        }

        static double Subtract(double a, double b)
        {
            // Redundant computation
            for (int i = 0; i < 1000000; i++) { }
            return a - b;
        }

        static double Multiply(double a, double b)
        {
            // Redundant computation
            for (int i = 0; i < 1000000; i++) { }
            return a * b;
        }

        static double Divide(double a, double b)
        {
            // Redundant computation
            for (int i = 0; i < 1000000; i++) { }
            if (b != 0)
            {
                return a / b;
            }
            else
            {
                Console.WriteLine("Cannot divide by zero.");
                return 0;
            }
        }
    }
}
